### PR TITLE
Add platform-userguide-test (New)

### PR DESCRIPTION
**Pull request by binli**
## Description

A new testcase to check if userguide package is installed

## Resolved issues

https://warthogs.atlassian.net/browse/OEX86-301

## Tests
```
=========[ Running job 4 / 4. Estimated time left (at least): 0:00:00 ]=========
--------------------[ miscellanea/platform-userguide-test ]---------------------
ID: com.canonical.certification::miscellanea/platform-userguide-test
Category: com.canonical.plainbox::miscellanea
... 8< -------------------------------------------------------------------------
Found the platform userguide package: oem-sutton-castro-doc
/tmp/nest-biwxewx3.43eea1fe921e92beedc650f60ad4d0ee2ba1e4925cb9bacfcbf157e09292eabb/platform_userguide_test passed!
------------------------------------------------------------------------- >8 ---
Outcome: job passed
Finalizing session that hasn't been submitted anywhere: checkbox-run-2024-06-07T07.29.16
==================================[ Results ]===================================
 ☑ : Resource to detect if dmi data is present
 ☑ : Collect information about hardware devices (DMI)
 ☑ : Collect information about installed system (lsb-release)
 ☑ : miscellanea/platform-userguide-test
```
```
=========[ Running job 4 / 4. Estimated time left (at least): 0:00:00 ]=========
--------------------[ miscellanea/platform-userguide-test ]---------------------
ID: com.canonical.certification::miscellanea/platform-userguide-test
Category: com.canonical.plainbox::miscellanea
... 8< -------------------------------------------------------------------------
Missing platform userguide packages!!!
/tmp/nest-nbc4iyl6.43eea1fe921e92beedc650f60ad4d0ee2ba1e4925cb9bacfcbf157e09292eabb/platform_userguide_test failed!
------------------------------------------------------------------------- >8 ---
Outcome: job failed
Finalizing session that hasn't been submitted anywhere: checkbox-run-2024-06-07T07.51.55
==================================[ Results ]===================================
 ☑ : Resource to detect if dmi data is present
 ☑ : Collect information about hardware devices (DMI)
 ☑ : Collect information about installed system (lsb-release)
 ☒ : miscellanea/platform-userguide-test
```

### CODE REVIEW

File: contrib/pc-sanity/bin/platform_userguide_test [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+failed()
+{
+    [ -n "$1" ] && echo "$1"
```
Comment (pseudocc) [inline contrib/pc-sanity/bin/platform_userguide_test:4]: ```[ -n "$1" ]```this is always true.

File: contrib/pc-sanity/bin/platform_userguide_test [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    echo "$0 failed!"
+    exit 1
+}
+
+passed()
+{
+    [ -n "$1" ] && echo "$1"
```
Comment (pseudocc) [inline contrib/pc-sanity/bin/platform_userguide_test:11]: `[ -n "$1" ]`this is always true.

File: contrib/pc-sanity/bin/platform_userguide_test [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+failed()
+{
+    [ -n "$1" ] && echo "$1"
+    echo "$0 failed!"
```
Comment (pseudocc) [inline contrib/pc-sanity/bin/platform_userguide_test:5]: How about using `>&2 echo` to print the errors to the `stderr` which makes more perfect sense.

File: contrib/pc-sanity/bin/platform_userguide_test (hunk 1 above)
Comment (pseudocc) [reply contrib/pc-sanity/bin/platform_userguide_test:4]: Although `$1` is correctly here, I prefer `$*`, what do you think?

**Review by pseudocc — COMMENTED** [review]
Two cents added.

File: contrib/pc-sanity/bin/platform_userguide_test (hunk 3 above)
Comment (binli) [reply contrib/pc-sanity/bin/platform_userguide_test:5]: thanks!

**Review by pseudocc — APPROVED** [review]

**Review by pseudocc — APPROVED** [review]
Thanks!

**Pull request merged**